conjur/credentials_from_file.py

```diff
@@ -29,40 +31,57 @@ def save(self, credential_data):
         Method that writes user data to a netrc file
         and updates permissions on the file
         """
-        with open(self.netrc_path, 'w') as netrc_file:
-            # pylint: disable=line-too-long
-            netrc_file.write(f"""machine {credential_data.machine}
-login {credential_data.login}
-password {credential_data.api_key}
-""")
+        if os.path.exists(self.netrc_path):
+            netrc_obj = netrc.netrc(self.netrc_path)
+            hosts = netrc_obj.hosts
+            hosts[credential_data.machine] = (credential_data.login, None, credential_data.api_key)
+            with open(DEFAULT_NETRC_FILE, "r+") as netrc_file:
+                ret = ""
+                for i, line in enumerate(str(netrc_obj).split('\n')):
+                    if line.strip().startswith('machine') and not i==0:
+                        ret += '\n'
+                    ret += line + '\n'
+                netrc_file.write(ret.replace('\t', ''))
+        else:
+            with open(self.netrc_path, "w+") as netrc_file:
+                credential_data.pretty_print(netrc_file, f"machine {credential_data.machine}")
+                credential_data.pretty_print(netrc_file, f"login {credential_data.login}")
+                credential_data.pretty_print(netrc_file, f"password {credential_data.api_key}")
```

Comment:
Sorry for the digging😬😬😬😬
why not 
```suggestion
            credential_data.save_to_file(self.netrc_path)
```